### DISCUSSION TITLE
Suggestions for Efficient Utilization of GPU Memory when using EWC Strategy

### DISCUSSION THREAD

**Discussion by piyushnags** [ContinualAI/avalanche#1425] [Q&A] [answered]
I've been trying to finetune a ViT model for Image Classification using the Replay and EWC Strategy in a class incremental scenario. The GPU usage increases after each experience (which makes sense since the Fisher matrix would need to be updated), and in my case, I run into an OOM issue when I'm about halfway done. I realize this is a situation where I need a GPU instance with more memory, but I was wondering if there are any techniques for efficient utilization of the GPU throughout this process.

On another note, I have 17 classes in my classification problem. The n_experiences parameter in the nc_benchmark function needs to be divisible by the total number of classes. Is there a way to work around this, since this makes it quite troublesome when dealing with prime numbers? This would actually solve my first problem since I'm training my model with 1 class per experience at the moment.

Thanks,
Piyush

**Comment by AntonioCarta** [comment, marked as answer]
There is an online mode for EWC.

For the benchmarks, `nc_benchmarks` allows you to control the exact size of each experience and the class order. [doc](https://avalanche-api.continualai.org/en/v0.3.1/generated/avalanche.benchmarks.generators.nc_benchmark.html#avalanche.benchmarks.generators.nc_benchmark)

**Reply by piyushnags** [reply]
Oh wow, no idea how I missed both of those parameters...
Thank you
